server/manager/src/main/java/org/apache/accumulo/manager/compaction/queue/CompactionJobPriorityQueue.java

```diff
@@ -307,11 +332,49 @@ private CjpqKey addJobToQueue(TabletMetadata tabletMetadata, CompactionJob job)
 
     var key = new CjpqKey(job);
     jobQueue.put(key, new CompactionJobQueues.MetaJob(job, tabletMetadata));
+    jobAges.computeIfAbsent(tabletMetadata.getExtent(), e -> Timer.startNew());
     return key;
   }
 
   public synchronized void clear() {
     jobQueue.clear();
     tabletJobs.clear();
+    jobAges.clear();
+
+    // TODO, what do we do here for the jobQueueTimer, if anything?
```

Comment:
Looking at how this is used its attempting to clean up memory for queues that are no longer used. This could be caused a by a change in config that causes new jobs to no longer be added. Could change this method to `clearIfInactive(Duration duration)` where it will only clear if nothing has been added in the passed in duration.  The impl could scan `jobAges.values()` and find the min elapsed time and compare that to the passed in duration.

The code that calls this clear method calls it every 5 mins so we could call `clearIfInactive(Duration.ofMinutes(10))`.